In [1]:
import numpy as np
import pandas as pd

In [2]:
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [3]:
sns.set()

In [5]:
data = pd.read_csv('medals.csv')

In [7]:
data.head()

Year      City       Sport      Discipline  NOC       Event Event gender  \
0  1924  Chamonix     Skating  Figure skating  AUT  individual            M   
1  1924  Chamonix     Skating  Figure skating  AUT  individual            W   
2  1924  Chamonix     Skating  Figure skating  AUT       pairs            X   
3  1924  Chamonix   Bobsleigh       Bobsleigh  BEL    four-man            M   
4  1924  Chamonix  Ice Hockey      Ice Hockey  CAN  ice hockey            M   

    Medal  
0  Silver  
1    Gold  
2    Gold  
3  Bronze  
4    Gold

In [12]:
data.dropna(axis=0,inplace=True)

In [24]:
data_med = data.groupby(['Sport'],as_index=False).count()

In [25]:
data_med

Sport  Year  City  Discipline   NOC  Event  Event gender  Medal
0    Biathlon   162   162         162   162    162           162    162
1   Bobsleigh   133   133         133   133    133           133    133
2     Curling    21    21          21    21     21            21     21
3  Ice Hockey    69    69          69    69     69            69     69
4        Luge   108   108         108   108    108           108    108
5     Skating   758   758         758   758    758           758    758
6      Skiing  1060  1060        1060  1060   1060          1060   1060

In [41]:
data['Medal'].value_counts()

Gold      774
Silver    773
Bronze    764
Name: Medal, dtype: int64

## ANALYSIS RESULT

The dataset is an olymic dataset with 2313 rows and 8 columns. 3 rows containing Nan values were dropped leaving me with 2311 rows.

From the above analysis done on the medals dataset, it can be seen that skiing as a sport has the highest number of medals compared to curling which has the lowest.

Also, as the year moves forward, the number of medals won increases as well. The data seems to be concerned with the medal won by each individual and which sport the medal was won playing. 

There are 67 different events accross the different sports in the dataset. Norway is the most represented country with 280 participants and Romania, Denmark, Uzbekistan, Slovakia, Latvia and New Zealand the leat represented countries with 1 participant each. 

Furthermore, there are 15 unique disciplines with speed skating having the highest athletes and Skeleton the lowedst with 455 and 18 respectively.

There are 17 cities represented in the dataset. This means, the events took place accross seventeen different cities with each having its own number of attendees and Turin being the cities with the highest numbers of attendees with 252 and Chamonix with 49 attendees.

According to the dataset, 774 individuals won the gold medal while 773 won silver and 764 won bronze medals respectively.